Integration/LdapAuthIntegration.php

```diff
@@ -122,14 +126,24 @@ public function authCallback($settings = [], $parameters = [])
             $ldap = new LdapClient($hostname, $port, $ldapVersion, $ssl, $startTls);
 
             try {
-                $userKey = $settings['user_key'];
-                $query = $settings['user_query'];
+                if ($isactivedirectory) {
+                    $patterns = array();
+                    $patterns[0] = '/[,]/';
+                    $patterns[1] = '/DC=|dc=|d=/';
+                    $patterns[2] = '/\w{1,2}=\w+\./'; // delete all other common names and attributes
+                    $replacements = array();
+                    $replacements[0] = '.'; // the only actual replacement
+                    $parsed_dn = preg_replace($patterns, $replacements, $base_dn);
+                    $dn = "$login@$parsed_dn";
+                } else {
+                    $dn = "$userKey=$login,$base_dn";
+                }
 
-                $dn = "$userKey=$login,$base_dn";
-                $password = $parameters['password'];
+                $userquery = "$userKey=$login";
+                $query = "(&($userquery)$query)"; // original $query already has brackets!
```

Comment:
Overall, it is always about where to give control to the person using the plugin. Perhaps, the main difficulty for us is to imagine what someone understands who is using the plugin for the first time. 

**AD parameters**
I think it is better to have these two parameters where one of them is exclusively to enable Active Directory and the other is for defining the logon domain. Just to prevent that someone who mistakingly entered `ldap_auth_activedirectory_domain` does not have to confirm that they actually want `isactivedirectory = true`

**Filter queries**
By exposing the actual user query to `ldap_auth_user_query` we will have a few disadvantages in my opinion:
1. Complexity: An admin has to understand that %u is a placeholder and this will need us to add more clarification. 
2. `ldap_auth_username_attribute` will get redundant when used in the query parameter. Also, I liked this parameter as it is very easy to understand what they need to enter there.
3.  Being able to write multiple specify multiple queries by using this syntax: 
`(query1)(query2)...(queryX)` seemed intuitive to me and is also reflected in the README

If couldn't convince you to keep it my way I will of course change the code accordingly.